src/patient-registration/form-manager.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  PatientIdentifier,
+  FormValues,
+  PatientIdentifierType,
+  AttributeValue,
+  PatientUuidMapType,
+  Patient,
+} from './patient-registration-helper';
+import { generateIdentifier } from './patient-registration.resource';
+import { ConfigObject, interpolateString } from '@openmrs/esm-framework';
+
+export default class FormManager {
+  static createIdentifiers(
+    values: FormValues,
+    patientUuidMap: object,
+    identifierTypes: Array<PatientIdentifierType>,
+    abortController: AbortController,
+    location: string,
+  ) {
+    const identifiers: Array<Promise<PatientIdentifier>> = identifierTypes.map(type => {
+      const idValue = values[type.fieldName];
+      if (idValue) {
+        return Promise.resolve({
+          uuid: patientUuidMap[type.fieldName] ? patientUuidMap[type.fieldName].uuid : undefined,
+          identifier: idValue,
+          identifierType: type.uuid,
+          location: location,
+          preferred: type.isPrimary,
+        });
+      } else if (type.autoGenerationSource) {
+        return new Promise((resolve, _) => {
+          generateIdentifier(type.autoGenerationSource.uuid, abortController).then(response => {
+            resolve({
+              // is this undefined?
+              uuid: undefined,
+              identifier: response.data.identifier,
+              identifierType: type.uuid,
+              location: location,
+              preferred: type.isPrimary,
+            });
+          });
+        });
+      }
+    });
+
+    return Promise.all(identifiers);
+  }
+
+  static populateAddressValues(values: FormValues, initialAddressFieldValues: Object): Record<string, string> {
```

Comment:
What's `Object`? You mean `Record<string, any>`?

Reply:
yea, i need to create a new type out of this - the return type never existed until i started refactoring